data-table/table.js

```diff
@@ -361,18 +368,104 @@ function createEntryCell(config, row, rowIndex, colIndex) {
                 option.innerHTML = config.datumConfig.values[fieldNum][i];
                 select.appendChild(option);
             }
-            label.appendChild(select);
+            field = select;
         } else {
             /**
              * FIXME: This error handling could be improved. Maybe a try-catch block?
              */
-            let errorText = "Cell field datatype not supported.";
-            throw errorText;
+            throw String("Cell field datatype not supported.");
         }
+
+        if (cellFieldHasCallback(config, fieldNum)) {
+            createCallbackListener(config, cell, field, fieldNum)
+        }
+
+        label.appendChild(field);
         cell.appendChild(label);
     }
 }
 
+/**
+ * Checks if a given field has a callback associated with it
+ * @param {object} config   - Table configuration object
+ * @param {Number} fieldNum - Index of the field within the cell
+ * @returns {boolean}       - Returns True if field has a callback
+ */
+function cellFieldHasCallback(config, fieldNum) {
+    return config.datumConfig.callbacks[fieldNum] !== undefined && config.datumConfig.callbacks[fieldNum] !== "None";
+}
+
+/**
+ * Creates a callback listener, distinguishing between the type of field to which the callback applies
+ * @param {object} config                               - Table configuration object
+ * @param {HTMLTableDataCellElement} cell               - Cell within the table
+ * @param {HTMLInputElement|HTMLSelectElement} field    - The field to which the callback applies
+ * @param {Number} fieldNum                             - Index of the field within the cell
+ * @returns {undefined}                                 - Doesn't return anything
+ */
+function createCallbackListener(config, cell, field, fieldNum) {
+
+    field.addEventListener("focusout", function () {
+        let fieldValue = null;
+
+        // Gets value of the field, depending on what type of field it is
+        if (field.classList.contains('cell-input')) {
+            fieldValue = [field.value.trim()]
+        } else if (field.classList.contains('cell-checkbox')) {
+            fieldValue = [field.checked]
+        } else if (field.classList.contains('cell-dropdown')) {
+            fieldValue = config.datumConfig.values[fieldNum][field.selectedIndex]
+        } else {
+            throw String('Field has no class');
+        }
+
+        let callbackCode = Reflect.apply(config.datumConfig.callbacks[fieldNum], config.datumConfig.callbacks[1], [fieldValue]);
+        handleCallbackReturn(config, cell, fieldNum, callbackCode);
+    })
+}
+
+/**
+ * Translates the return value of a callback to function calls
+ * @param {object} config                   - Table configuration object
+ * @param {HTMLTableDataCellElement} cell   - Cell within the table
+ * @param {Number} fieldNum                 - Index of the field within the cell
+ * @param {String} callbackCode             - Client's response to the field's value
```

Comment:
I don't love that the callback returns just `Invalid` - can we tell the user _why_? e.g. "Negative numbers are not allowed"? Perhaps any string should be shown directly as the error message, and null means everything was okay?

Whatever we decide, make sure it's documented both in code (where the config is initialized) and eventually, in the README

Reply:
Okay - I think I could use some conversation around this to make sure I'm understanding correctly. For now, I'm marking this down as a topic to bring up at our next meeting (and adding a FIXME, in the meantime).